Common helper function for Infernity patches.

CalculateTextWidth() from Infernity by quendel, extracted by sergi4ua.

Rebased on devilutionX 0.3.1.

diff --git a/Source/engine.cpp b/Source/engine.cpp
index 7911ceb..77d80f4 100644
--- a/Source/engine.cpp
+++ b/Source/engine.cpp
@@ -571,6 +571,24 @@ L_ODD:
 }
 // 69BEF8: using guessed type int light_table_index;
 
+void DrawSolidRectangle(int x0, int dx, int y0, int dy, int color) {
+	char* WorkingSurface = (char*)gpBuffer;
+	for (int x = x0; x < x0 + dx; x++) {
+		for (int y = y0; y < y0 + dy; y++) {
+			WorkingSurface[y*768+x] = color;
+		}
+	}
+}
+
+int CalculateTextWidth(char* s)
+{
+	int l = 0;
+	while (*s) {
+		l += fontkern[fontframe[gbFontTransTbl[*s++]]] + 1;
+	}
+	return l;
+}
+
 void __fastcall CelDecodeLightOnly(int sx, int sy, BYTE *pCelBuff, int nCel, int nWidth)
 {
 	int nDataSize;
diff --git a/Source/engine.h b/Source/engine.h
index 7c5352b..4a429e9 100644
--- a/Source/engine.h
+++ b/Source/engine.h
@@ -18,6 +18,8 @@ void __fastcall CelDrawHdrOnly(int sx, int sy, BYTE *pCelBuff, int nCel, int nWi
 void __fastcall CelDecodeHdrOnly(BYTE *pBuff, BYTE *pCelBuff, int nCel, int nWidth, int always_0, int dir);
 void __fastcall CelDecDatLightOnly(BYTE *pDecodeTo, BYTE *pRLEBytes, int nDataSize, int nWidth);
 void __fastcall CelDecDatLightTrans(BYTE *pDecodeTo, BYTE *pRLEBytes, int nDataSize, int nWidth);
+void DrawSolidRectangle(int x0, int dx, int y0, int dy, int color);
+int CalculateTextWidth(char* s);
 void __fastcall CelDecodeLightOnly(int sx, int sy, BYTE *pCelBuff, int nCel, int nWidth);
 void __fastcall CelDecodeHdrLightOnly(int sx, int sy, BYTE *pCelBuff, int nCel, int nWidth, int always_0, int dir);
 void __fastcall CelDecodeHdrLightTrans(BYTE *pBuff, BYTE *pCelBuff, int nCel, int nWidth, int always_0, int dir);
